oat-sa/tao-core
# Feature/RFE-1147/Integration/Apply the Solar Design

### Related to: https://oat-sa.atlassian.net/browse/RFE-1147

Requires: 
 - [ ] https://github.com/oat-sa/extension-tao-itemqti/pull/2505
 - [ ] https://github.com/oat-sa/extension-tao-testqti/pull/2473

### Summary

Apply the Solar Design to the backoffice. This is the integration pull request, changes have been already reviewed through intermediate pull requests:
- https://github.com/oat-sa/tao-core/pull/4006
- https://github.com/oat-sa/tao-core/pull/4011
- https://github.com/oat-sa/tao-core/pull/4015
- https://github.com/oat-sa/tao-core/pull/4018
- https://github.com/oat-sa/tao-core/pull/4019
- https://github.com/oat-sa/tao-core/pull/4022
- https://github.com/oat-sa/tao-core/pull/4024
- https://github.com/oat-sa/tao-core/pull/4025
- https://github.com/oat-sa/tao-core/pull/4026
- https://github.com/oat-sa/tao-core-ui-fe/pull/592

![image](https://github.com/oat-sa/tao-core/assets/1500098/c02fda2b-3f5d-4ecc-8f25-e4b95eef6d03)

### Details

To activate the feature, you need the feature flag `FEATURE_FLAG_SOLAR_DESIGN_ENABLED`.
```sh
php index.php 'oat\tao\scripts\tools\FeatureFlag\FeatureFlagTool' -s FEATURE_FLAG_SOLAR_DESIGN_ENABLED -v true
```

Once activated, the CSS class `solar-design` is added to the body element of the page. Then, the stylesheet automatically applies.

The Solar Design has been partially replicated in the CSS variables. For compatibility reasons, some have been simplified.

The backoffice is now following the Solar Design.

### How to test

- check out the branch: `git checkout -t origin/feature/RFE-1147/integration`
- activate the feature flag: `php index.php 'oat\tao\scripts\tools\FeatureFlag\FeatureFlagTool' -s FEATURE_FLAG_SOLAR_DESIGN_ENABLED -v true`
- log in to TAO backoffice
- look at the pages, compare with the UX mockups
